python/pyarrow/_hdfsio.pyx

```diff
@@ -20,16 +20,8 @@
 
 # cython: language_level = 3
 
-import re
-
-from pyarrow.lib cimport check_status, _Weakrefable, NativeFile
-from pyarrow.includes.common cimport *
-from pyarrow.includes.libarrow cimport *
+from pyarrow.lib cimport check_status
 from pyarrow.includes.libarrow_fs cimport *
-from pyarrow.lib import frombytes, tobytes, ArrowIOError
-
-
-_HDFS_PATH_RE = re.compile(r'hdfs://(.*):(\d+)(.*)')
 
 
 def have_libhdfs():
```

Comment:
If this is the only remaining thing in this file, we can maybe move it elsewhere (maybe io.pxi?)